ada/extensions/src/libadalang-helpers.ads

```diff
@@ -0,0 +1,82 @@
+with Ada.Containers.Vectors;
+with Ada.Strings.Unbounded; use Ada.Strings.Unbounded;
+
+with GNATCOLL.Opt_Parse;
+
+with Libadalang.Analysis; use Libadalang.Analysis;
+
+package Libadalang.Helpers is
+
+   package Unit_Vectors is new
+     Ada.Containers.Vectors (Positive, Analysis_Unit);
+
+   --  This package is a convenient and easy way to create an application based
+   --  on Libadalang, with out of the box facilities such as:
+   --
+   --  * Automatic handling of project files, including an option to process
+   --    all the files of the project, and handling of scenario variables.
+   --
+   --  * Automatic command line option parser, with descriptive help.
+   --
+   --  As a user, all you have to do is to provide a ``Process_Unit``
+   --  procedure, that will process one unit out of the set of units
+   --  to process.
+   generic
+      Description : String;
+
+      with procedure Process_Context
+        (Ctx : Analysis_Context; Units : Unit_Vectors.Vector) is null;
+      --  This procedure will be called once after all units have been parsed
+
+      with procedure Process_Unit
+        (Unit : Analysis_Unit) is null;
+      --  This procedure will be called once right after an unit is parsed
+
+   package App is
+
+      --  This package contains the arguments parser for your app. It contains
+      --  the instantiation of the parser, and the set of default arguments, so
+      --  that you can also use their values in your app if needed.
+      package Args is
+         use GNATCOLL.Opt_Parse;
+
+         Parser : Argument_Parser := Create_Argument_Parser
+           (Help => Description);
+         --  Argument parser for your application. Supports a set of default
+         --  options. You can add your own on this parser.
+
+         package Charset is new Parse_Option
+           (Parser, "-C", "--charset", "Charset to use for source decoding",
+            Unbounded_String,
+            Default_Val => To_Unbounded_String ("iso-8859-1"));
+
+         package Scenario_Vars is new Parse_Option_List
+           (Parser, Short => "-X", Long => "--scenario-variables",
```

Comment:
Do we have to provide a long option? This one is non standard (I don’t think we have long options for scenario vars), and if we preserve it, it should probably be `--scenario-variable` (one option = one scenario variable).

Reply:
You're right. I'll check `GNATCOLL.Opt_Parse` to see if we have to provide the long or not.

Reply:
So, I checked in `Opt_Parse`, and the long flag is indeed not optional. I think the reasoning at the time is that all flags should have a long descriptive form.

We can change that if you think it's necessary (I told you Opt_Parse was not yet stable! :), personally since this is backwards compatible I don't really care. I'll remove the s for this PR and we can see later for the bigger project of removing long if you think it's worth it.

Reply:
Keeping the long form is fine for now, IMO. Thanks!